src/hotspot/share/classfile/stackMapTable.cpp

```diff
@@ -29,38 +29,50 @@
 #include "oops/oop.inline.hpp"
 #include "runtime/handles.inline.hpp"
 
-StackMapTable::StackMapTable(StackMapReader* reader, StackMapFrame* init_frame,
-                             u2 max_locals, u2 max_stack,
-                             char* code_data, int code_len, TRAPS) {
-  _code_length = code_len;
+StackMapTable::StackMapTable(StackMapReader* reader, TRAPS) {
+  _code_length = reader->code_length();
   _frame_count = reader->get_frame_count();
   if (_frame_count > 0) {
-    _frame_array = NEW_RESOURCE_ARRAY_IN_THREAD(THREAD,
-                                                StackMapFrame*, _frame_count);
-    StackMapFrame* pre_frame = init_frame;
-    for (int32_t i = 0; i < _frame_count; i++) {
-      StackMapFrame* frame = reader->next(
-        pre_frame, i == 0, max_locals, max_stack,
-        CHECK_VERIFY(pre_frame->verifier()));
-      _frame_array[i] = frame;
-      int offset = frame->offset();
-      if (offset >= code_len || code_data[offset] == 0) {
-        frame->verifier()->verify_error(
-            ErrorContext::bad_stackmap(i, frame),
-            "StackMapTable error: bad offset");
-        return;
+    _frame_array = new GrowableArray<StackMapFrame*>();
```

Comment:
I think if you give GrowableArray() an initial capacity of _frame_count, that should be the right count unless you have assert_unset_field stackmap modifiers, in which case you'll use less but only a little bit less.  The default size is 2 of GrowableArray.